pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,223 @@ func TestAccessTokenFile(t *testing.T) {
 	var tokenInFile, _ = NewAccessTokenFile(config)
 
 	Convey("Read", t, func() {
-		Convey("Returns true if Data output as expected", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
+			Convey("Finishes without raising an error", func() {
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("When running Read method", func() {
+				dataExpected, err := tokenInFile.Read()
 
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Returns the data that was written", func() {
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 
-		Convey("Returns error if Data is nil", func() {
+		Convey("Given an access token's data is empty", func () {
 			tokenInFile.Data = nil
-			_, err := tokenInFile.Read()
 
-			So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+			Convey("When running the Read method", func() {
+				_, err := tokenInFile.Read()
+
+				Convey("Raises an error that the data is empty", func() {
+					So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+				})
+			})
 		})
 	})
 
 	Convey("Write", t, func() {
-		Convey("Returns no error if Data input is not nil", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
-			err := tokenInFile.Write(dataActual)
 
-			So(err, ShouldEqual, nil)
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
+			Convey("When running the Write method", func() {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Checks that the file exists in the path defined", func (){
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite1")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("When running Read method", func() {
+					dataExpected, _ := tokenInFile.Read()
+
+					// Confirm data was written
+					Convey("Returns the data the was written to the file", func() {
+						eq := reflect.DeepEqual(dataActual, dataExpected)
+						So(eq, ShouldEqual, true)
+					})
+				})
```

Comment:
I'd rather have this as:
```
Convey("Writes the data to the file", func() {
	// TODO: read the content with `os` methods (not with `tokenInFile`)

	eq := reflect.DeepEqual(dataActual, dataExpected)
	So(eq, ShouldEqual, true)
})
```

Reply:
This way it explains what we are testing and we won;'t fail on `tokenInFile.Read()` errors. We are pin-pointing the test to what is tested without noises

Reply:
I thought TODO is something that is awaiting implementation. If I remove `dataExpected, _ := tokenInFile.Read()` I won't be able to compare if it was the proper result. Suggest putting all  commented out?

Reply:
TODO by you now :)

Reply:
what i meant is that we should read with `os` methods and not with our methods. this is because it's better to assume `os` methods work rather than our methods